aws/resource_aws_dx_gateway_association_proposal.go

```diff
@@ -18,14 +20,47 @@ func resourceAwsDxGatewayAssociationProposal() *schema.Resource {
 			State: schema.ImportStatePassthrough,
 		},
 
+		CustomizeDiff: customdiff.Sequence(
+			// Accepting the proposal with overridden prefixes changes the returned RequestedAllowedPrefixesToDirectConnectGateway value (allowed_prefixes attribute).
+			// We only want to force a new resource if this value changes and the current proposal state is "requested".
+			customdiff.ForceNewIf("allowed_prefixes", func(d *schema.ResourceDiff, meta interface{}) bool {
+				conn := meta.(*AWSClient).dxconn
+
+				proposal, err := describeDirectConnectGatewayAssociationProposal(conn, d.Id())
+				if err != nil {
+					log.Printf("[ERROR] Error reading Direct Connect Gateway Association Proposal (%s): %s", d.Id(), err)
+					return false
+				}
+
+				if proposal != nil && aws.StringValue(proposal.ProposalState) == directconnect.GatewayAssociationProposalStateRequested {
```

Comment:
looks like we can just return the results of the final condition here; dropping lines 35-39 for the following: 

```go
return proposal != nil && aws.StringValue(proposal.ProposalState) == directconnect.GatewayAssociationProposalStateRequested
```